edk2toollib/uefi/edk2/path_utilities.py

```diff
@@ -117,19 +119,19 @@ def GetAbsolutePathOnThisSystemFromEdk2RelativePath(self, relpath, log_errors=Tr
 
         if relpath is None:
             return None
-        relpath = relpath.replace("/", os.sep)
+        relpath = relpath.replace("/", os.sep).replace('\\', os.sep)
```

Comment:
instead of the replace logic should you just use https://docs.python.org/3/library/os.path.html#os.path.normpath

should this be used on all input strings that are going to be filesystem specific?

Reply:
That works on Windows, but not on Linux 😢. Linux thinks `\` is a valid character for file and folder names, so it assumes you have a folder or file named `foo\bar` and won't clean them up:
```console
root@tryit-active:~# mkdir foo                                                                                                                         
root@tryit-active:~# mkdir foo\\bar                                                                                                                    
root@tryit-active:~# ls -l                                                                                                                             
total 0                                                                                                                                                
drwxr-xr-x 1 root root 0 Jul 14 02:14  foo                                                                                                             
drwxr-xr-x 1 root root 0 Jul 14 02:15 'foo\bar' 
```

If some functions are expected to return an edk2 path that uses `/` regardless of OS, I don't have a great solution. Returning os.sep from everything *except sometimes `/`* isn't very user-friendly. Forcing `/` as the separator in all cases might cause a bunch of trouble for users? Or we could not make a guarantee and say the bug is in the tool that was expecting the separators to all be the same (the one I ran into was the guid check plugin).